x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -277,132 +421,157 @@ private static XContentBuilder getDynamicTemplates(XContentBuilder builder) thro
             .endArray();
     }
 
-    /**
-     * Creates the rollup mapping properties from the provided {@link RollupActionConfig}.
-     */
-    private static XContentBuilder getProperties(XContentBuilder builder, RollupActionConfig config) throws IOException {
-        builder.startObject("properties");
-
-        RollupActionGroupConfig groupConfig = config.getGroupConfig();
-        RollupActionDateHistogramGroupConfig dateHistogramConfig = groupConfig.getDateHistogram();
-        String dateField = dateHistogramConfig.getField();
-        String dateIntervalType = dateHistogramConfig.getIntervalTypeName();
-        String dateInterval = dateHistogramConfig.getInterval().toString();
-        String tz = dateHistogramConfig.getTimeZone() != null
-            ? dateHistogramConfig.getTimeZone()
-            : RollupActionDateHistogramGroupConfig.DEFAULT_TIMEZONE;
-
-        builder.startObject(dateField)
-            .field("type", DateFieldMapper.CONTENT_TYPE)
-            .startObject("meta")
-            .field(dateIntervalType, dateInterval)
-            .field(RollupActionDateHistogramGroupConfig.CalendarInterval.TIME_ZONE, tz)
-            .endObject()
-            .endObject();
-
-        HistogramGroupConfig histogramGroupConfig = groupConfig.getHistogram();
-        if (histogramGroupConfig != null) {
-            for (String field : histogramGroupConfig.getFields()) {
-                builder.startObject(field)
-                    .field("type", NumberFieldMapper.NumberType.DOUBLE.typeName())
-                    .startObject("meta")
-                    .field(HistogramGroupConfig.INTERVAL, String.valueOf(histogramGroupConfig.getInterval()))
-                    .endObject()
-                    .endObject();
+    private void createRollupIndex(
+        String rollupIndexName,
+        IndexMetadata sourceIndexMetadata,
+        String mapping,
+        RollupAction.Request request,
+        ActionListener<AcknowledgedResponse> listener
+    ) {
+        CreateIndexClusterStateUpdateRequest createIndexClusterStateUpdateRequest = new CreateIndexClusterStateUpdateRequest(
+            "rollup",
+            rollupIndexName,
+            rollupIndexName
+        ).settings(
+            /*
+             * When creating the rollup index, we copy the index.number_of_shards from source index,
+             * and we set the index.number_of_replicas to 0, to avoid replicating the temp index.
+             * We will set the correct number of replicas later.
+             */
+            Settings.builder()
+                .put(IndexMetadata.INDEX_HIDDEN_SETTING.getKey(), true)
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, sourceIndexMetadata.getNumberOfShards())
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .put(IndexMetadata.INDEX_ROLLUP_STATUS.getKey(), IndexMetadata.RollupTaskStatus.STARTED)
+                .build()
```

Comment:
Maybe we should set the refresh_interval to -1 here. The default interval tries to make data visible while writing at the cost of less efficient writes.

Reply:
Also - I wonder if we should force_merge after we're done writing. It doesn't have to be now, but it should make the target index more efficient to query and we're not going to write to it again.

Reply:
Fixed in [0ff80dc](https://github.com/elastic/elasticsearch/pull/85708/commits/0ff80dc4792a1caa76bbb44566e9f4f6e5c4774d)

- Set refresh_interval to -1
- Force merge index in the end